vulnerabilities/importers/elixir_security.py

```diff
@@ -0,0 +1,139 @@
+
+
+class ElixirSecurityDataSource(GitDataSource):
+    def __enter__(self):
+        super(ElixirSecurityDataSource, self).__enter__()
+
+        if not getattr(self, "_added_files", None):
+            self._added_files, self._updated_files = self.file_changes(
+                recursive=True, file_ext="yml", subdir="./packages"
+            )
+        self.pkg_manager_api = HexVersionAPI()
+        self.set_api(self.collect_packages())
+
+    def set_api(self, packages):
+        asyncio.run(self.pkg_manager_api.load_api(packages))
+
+    def updated_advisories(self) -> Set[Advisory]:
+        files = self._updated_files
+        advisories = []
+        for f in files:
+            processed_data = self.process_file(f)
+            if processed_data:
+                advisories.append(processed_data)
+        return self.batch_advisories(advisories)
+
+    def added_advisories(self) -> Set[Advisory]:
+        files = self._added_files
+        advisories = []
+        for f in files:
+            processed_data = self.process_file(f)
+            if processed_data:
+                advisories.append(processed_data)
+        return self.batch_advisories(advisories)
+
+    def collect_packages(self):
+        packages = set()
+        files = self._updated_files.union(self._added_files)
+        for f in files:
+            with open(f) as file:
+                data = yaml.safe_load(file)
+                if data.get("package"):
+                    packages.add(data["package"])
+
+        return packages
+
+    def generate_all_version_list(self, pkg_name):
+        if not getattr(self, 'pkg_manager_api', None):
+            self.pkg_manager_api = HexVersionAPI()
+        version_list = self.pkg_manager_api.get(
+            pkg_name)
+        return version_list
+
+    def get_versions_from_range(self, version_list, pkg_name):
+        pkg_versions = []
+        all_version_list = self.generate_all_version_list(pkg_name)
+        if version_list is None:
+            return
+        version_ranges = {RangeSpecifier(r) for r in version_list}
+        for version in all_version_list:
+            if any([version in v for v in version_ranges]):
+                pkg_versions.append(version)
+        return pkg_versions
+
+    def process_file(self, path):
+        with open(path) as f:
+            yaml_file = yaml.safe_load(f)
+            pkg_name = yaml_file["package"]
+            safe_pkg_versions = []
+            if yaml_file.get("unaffected_versions"):
+                safe_pkg_versions = self.get_versions_from_range(
+                    yaml_file["patched_versions"] +
+                    yaml_file["unaffected_versions"],
+                    pkg_name,
+                )
+            else:
+                safe_pkg_versions = self.get_versions_from_range(
+                    yaml_file["patched_versions"], pkg_name
+                )
+
+            cve_id = "CVE-"+str(yaml_file["cve"])
```

Comment:
Why  the str cast ?

Reply:
without str cast there was an error like<code>cannot add None type to string</code> .So I did it. Don't know if there is another way.

Reply:
Nooo ... that makes the str cast even worse

In case `yaml_file["cve"] = None` , obviously it means the provided advisory has no  CVE, which is what our data should reflect but instead now, due to the cast it's gonna become `CVE-None`. 

I suggest something like : 
```Python
if yaml_file.get('cve') : 
    cve_id = 'CVE-'+yaml_file['cve']
else : 
    cve_id = ""

```

Reply:
Ok